cocos/audio/android/jni/cddandroidAndroidJavaEngine.cpp

```diff
@@ -36,282 +39,463 @@ THE SOFTWARE.
 // Java class
 #define  CLASS_NAME "org/cocos2dx/lib/Cocos2dxHelper"
 
-namespace CocosDenshion {
-    namespace android {
-
-        bool AndroidJavaEngine::getJNIStaticMethodInfo(cocos2d::JniMethodInfo &methodinfo,
-                                                       const char *methodName,
-                                                       const char *paramCode) {
-            return cocos2d::JniHelper::getStaticMethodInfo(methodinfo,
-                                                           CLASS_NAME,
-                                                           methodName,
-                                                           paramCode);
+using namespace cocos2d::experimental;
+using namespace CocosDenshion::android;
+
+static inline bool getJNIStaticMethodInfo(cocos2d::JniMethodInfo &methodinfo,
+    const char *methodName,
+    const char *paramCode) {
+    return cocos2d::JniHelper::getStaticMethodInfo(methodinfo,
+        CLASS_NAME,
+        methodName,
+        paramCode);
+}
+
+AndroidJavaEngine::AndroidJavaEngine()
+    : _implementBaseOnAudioEngine(false)
+    , _effectVolume(1.f)
+{
+    char sdk_ver_str[PROP_VALUE_MAX] = "0";
+    auto len = __system_property_get("ro.build.version.sdk", sdk_ver_str);
+    if (len > 0)
+    {
+        auto sdk_ver = atoi(sdk_ver_str);
+        __android_log_print(ANDROID_LOG_DEBUG, "cocos2d", "android build version:%d", sdk_ver);
+        if (sdk_ver == 21)
```

Comment:
Why is this sdk_ver == 21? Why not  sdk_ver  >= 21?
Does Android API 22 Android 5.1 LOLLIPOP_MR1 fix the [play effect lagged] issue?